analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+    cluster_columns = TRUE,
+    show_column_names = TRUE,
+    column_names_gp = grid::gpar(fontsize = 8),
+    heatmap_legend_param = list(
+      title = "Mean expression"
+    )
+  )
+
+```
+
+- Here it looks like the tumor cells have high expression of `EWS-FLI1` upregulated gene sets and low expression of `EWS-FLI1` repressed targets. 
+This suggests these cells are `EWS-high`. 
+- This same pattern is seen in cells labeled as "smooth muscle cell" and "stromal cell". 
+There may even be some slight expression of `EWS-FLI1` upregulated gene sets in "muscle cell", "Unknown", and some of the T cell populations. 
+Some of these cells are probably mistakenly labeled and should be tumor cells that are "EWS-high".
+- Fibroblasts and chondroctyes have low expression of `EWS-FLI1` upregulated gene sets and high expression of `EWS-FLI1` repressed targets. 
+Many of the genes in the `EWS-FLI1` repressed targets are also genes upregulated in fibroblasts so I'm sure we have a mix of tumor and fibroblasts in there.
+
+Because we have some normal cell types that are showing gene signatures expected to be specific to tumor cells, we should use the AUC values to help pull out any remaining cells that should be tumor cells. 
+We can also use the up and down gene sets to distinguish EWS-high and EWS-low tumor cells. 
+
+Let's pick some gene sets where we see variation across cell types that we can use to define cutoffs: 
+
+- EWS-high gene sets: `aynaud-ews-targets` and `staege`
+- EWS-low gene sets: `wrenn-nt5e-genes` and `hallmark_EMT`
+
+```{r}
+genesets_of_interest = c(
+  "auc_aynaud-ews-targets",
+  "auc_staege",
+  "auc_wrenn-nt5e-genes",
+  "auc_hallmark_EMT"
+)
+
+cluster_density_plot(all_info_df, genesets_of_interest, "combined_lumped", "AUC")
+```
+
+We can use these plots to help define cutoffs of AUC values to use for re-labeling any normal cells that should be tumor cells. 
+We expect that the endothelial cells and macrophages have low AUC values so we can use these cell types as the background. 
+Any AUC values above that background should be classified as tumor. 
+
+With this in mind, let's use these cutoffs to define tumor cells:
+
+- `aynaud-ews-targets` > 0.025 AUC
+- `staege` > 0.025 AUC
+- `wrenn-nt5e-genes` > 0.1 AUC
+- `hallmark_EMT` > 0.05 AUC 
+
+Let's re-plot the density plots adding in a line for the AUC cutoffs we have chosen. 
+
+```{r}
+cutoffs = c(
+  "auc_aynaud-ews-targets" = 0.025,
+  "auc_staege" = 0.025,
+  "auc_wrenn-nt5e-genes" = 0.1,
+  "auc_hallmark_EMT" = 0.05
+)
+
+
+cutoffs |>
+    purrr::imap(\(cutoff, column){
+      plot_density(
+        all_info_df,
+        column,
+        "combined_lumped"
+      ) +
+        theme(text = element_text(size = 8)) +
+        geom_vline(xintercept = cutoff)
+    }) |>
+    patchwork::wrap_plots(ncol = 2)
+```
+
+I don't think these cutoffs are perfect, but I think they are reasonable. 
+Any cells that are to the left of these cutoffs will be classified as tumor. 
```

Comment:
do you mean to the right? greater than cutoff = tumor, correct?

Reply:
I'm laughing because I am notoriously bad at lefts and rights. So yes I mean to the right 😂